api/agent/drivers/docker/docker.go

```diff
@@ -54,16 +54,59 @@ func (r *runResult) Error() error   { return r.err }
 func (r *runResult) Status() string { return r.status }
 
 type DockerDriver struct {
-	conf     drivers.Config
-	docker   dockerClient // retries on *docker.Client, restricts ad hoc *docker.Client usage / retries
-	hostname string
-	auths    map[string]driverAuthConfig
-	pool     DockerPool
+	conf       drivers.Config
+	docker     dockerClient // retries on *docker.Client, restricts ad hoc *docker.Client usage / retries
+	hostname   string
+	auths      map[string]driverAuthConfig
+	pool       DockerPool
+	imageCache *Cache
 	// protects networks map
 	networksLock sync.Mutex
 	networks     map[string]uint64
 }
 
+func NewImageCleaner(dockerDriver *DockerDriver, context context.Context) error {
+	opts := docker.RemoveImageOptions{}
+	opts.Force = true
+	opts.NoPrune = false
+	opts.Context = context
+	ticker := time.NewTicker(time.Minute * 5)
```

Comment:
Nit: would you please change the '5' to a const variable.  `ImageCleanScanMinutes = 5` or whatever is stylistically consistent for this code, perhaps?